### DISCUSSION TITLE
This method cannot be used with flat config. Add your entries directly into the config array.

### DISCUSSION THREAD

**Discussion by gajus** [eslint/eslint#18470] [General]
```
 FAIL  src/rules/enforceStitchesTokens.test.ts [ src/rules/enforceStitchesTokens.test.ts ]
Error: This method cannot be used with flat config. Add your entries directly into the config array.
 ❯ assertEslintrcConfig ../../node_modules/.pnpm/eslint@9.2.0/node_modules/eslint/lib/linter/linter.js:1256:15
 ❯ Linter.defineRule ../../node_modules/.pnpm/eslint@9.2.0/node_modules/eslint/lib/linter/linter.js:2164:9
 ❯ RuleTester.run ../../node_modules/.pnpm/@typescript-eslint+rule-tester@7.9.0_@eslint+eslintrc@3.1.0_eslint@9.2.0_typescript@5.4.3/node_modules/@typescript-eslint/rule-tester/src/RuleTester.ts:400:18
 ❯ src/rules/enforceStitchesTokens.test.ts:6:12
      4| const ruleTester = new RuleTester();
      5|
      6| ruleTester.run('enforce-stitches-tokens', rule, {
       |            ^
      7|   invalid: [
      8|     {
```

```ts
import rule from './validateCronString.js';
import { RuleTester } from '@typescript-eslint/rule-tester';

const ruleTester = new RuleTester();

ruleTester.run('validate-cron-strings', rule, {
  invalid: [
    {
      code: `const cronWorkflows = [
        { cronSchedule: '/30 * * * *' },
      ];`,
      errors: [
        {
          messageId: 'invalidCronString',
        },
      ],
    },
  ],
  valid: [
    {
      code: `const cronWorkflows = [
        { cronSchedule: '*/30 * * * *' },
      ];`,
    },
  ],
});
```

What does this error mean and how should the code be refactored?

**Comment by aladdin-add** [comment]
> import { RuleTester } from '@typescript-eslint/rule-tester';

looks like you are not using eslint-rule-tester directly. Not sure if it's doing some different things (I'm not familiar with the tseslint rule-tester). /cc @JoshuaKGoldberg

**Reply by JoshuaKGoldberg** [reply]
Yeah, this isn't an ESLint issue - it's a typescript-eslint issue. We had to fork the rule-tester (https://github.com/typescript-eslint/typescript-eslint/pull/5750 -> https://github.com/typescript-eslint/typescript-eslint/pull/6777) so it's a lot of totally separate code paths.

**Reply by azat-io** [reply]
This problem seems to occur after upgrading to ESLint v9.
Do you know if there's any way to solve it now?

**Reply by aladdin-add** [reply]
tseslint v8 added eslint v9 supports, you can try it.

https://github.com/typescript-eslint/typescript-eslint/issues/8211

**Reply by azat-io** [reply]
I tried it. The problem is there.

I think the problem is described here:
https://github.com/typescript-eslint/typescript-eslint/issues/9017

**Comment by electrovir** [comment]
I ran into this while trying to use `@typescript-eslint/eslint-plugin`'s `rc-v8` tagged version, which I thought would work _better_ with ESLint v9 because `typescript-eslint` doesn't work at all unless its `rc-v8` version is used. Reverting `@typescript-eslint/eslint-plugin` back to v7 fixed this issue for me.
